spec/models/audit_activity/risk_assessment/risk_assessment_added_spec.rb

```diff
@@ -0,0 +1,198 @@
+  subject(:activity) do
+    described_class.create!(
+      source: UserSource.new(user: risk_assessment.added_by_user),
+      investigation: risk_assessment.investigation,
+      metadata: metadata
+    )
+  end
+
+  let(:risk_assessment) { create(:risk_assessment, trait, assessed_by_team: assessed_by_team, assessed_by_business: assessed_by_business, assessed_by_other: assessed_by_other, products: products) }
+  let(:trait) { :without_file }
+  let(:assessed_by_team) { nil }
+  let(:assessed_by_business) { nil }
+  let(:assessed_by_other) { "test" }
+  let(:products) { [build(:product)] }
+  let(:metadata) { described_class.build_metadata(risk_assessment) }
+
+  describe "#metadata" do
+    context "when the metadata is in the current format" do
+      it "returns the metadata" do
+        expect(activity.metadata).to eq(activity.read_attribute(:metadata))
+      end
+    end
+
+    context "when the activity was generated prior to the metadata structure revision" do
+      let(:trait) { :with_file }
+      let(:metadata) do
+        {
+          risk_assessment_id: risk_assessment.id,
+          assessed_on: risk_assessment.assessed_on,
+          risk_level: risk_assessment.risk_level,
+          custom_risk_level: risk_assessment.custom_risk_level,
+          assessed_by_team_id: risk_assessment.assessed_by_team_id,
+          assessed_by_business_id: risk_assessment.assessed_by_business_id,
+          assessed_by_other: risk_assessment.assessed_by_other,
+          details: risk_assessment.details,
+          product_ids: risk_assessment.product_ids
+        }
+      end
+
+      # rubocop:disable RSpec/ExampleLength
+      it "returns a Hash in the new format" do
+        expect(activity.metadata).to match({
+          "risk_assessment" => hash_including({
+            "id" => risk_assessment.id,
+            "investigation_id" => risk_assessment.investigation_id,
+            "assessed_on" => "2020-07-20",
+            "assessed_by_team_id" => risk_assessment.assessed_by_team_id,
+            "assessed_by_business_id" => risk_assessment.assessed_by_business_id,
+            "assessed_by_other" => risk_assessment.assessed_by_other,
+            "details" => risk_assessment.details,
+            "custom_risk_level" => risk_assessment.custom_risk_level,
+            "added_by_user_id" => risk_assessment.added_by_user_id,
+            "added_by_team_id" => risk_assessment.added_by_team_id,
+            "risk_level" => risk_assessment.risk_level,
+            "product_ids" => risk_assessment.product_ids
+          })
+        })
+      end
+      # rubocop:enable RSpec/ExampleLength
+
+      context "when the risk assessment has not been subsequently updated" do
+        it "returns the file blob metadata" do
+          expect(activity.metadata["risk_assessment"]["risk_assessment_file"]).to match(hash_including(risk_assessment.risk_assessment_file.blob.attributes.except("created_at", "updated_at")))
```

Comment:
A trick I use to get around the string timestamps.
```suggestion
          expect(activity.metadata["risk_assessment"]["risk_assessment_file"]).to match(hash_including(JSON.parse(risk_assessment.risk_assessment_file.blob.attributes.to_json)))
```